### PR TITLE
Adds workflow job template form, and details

### PR DESCRIPTION
**Pull request by AlexSCorey**
Adds Workflow Job Template Form and Workflow Job Template Detail View

### CODE REVIEW

File: cypress/e2e/awx/resources/workflowJobTemplates.cy.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    // Name
+    cy.typeInputByLabel(/^Name$/, jtName);
+
+    // Description
+    cy.typeInputByLabel(/^Description$/, 'this is a description');
+
+    cy.selectDropdownOptionByLabel(/^Labels$/, label.name.toString(), true);
+
+    // Job tags
+    cy.contains('.pf-c-form__label-text', /^Job tags$/)
```
Comment (jamestalton) [inline cypress/e2e/awx/resources/workflowJobTemplates.cy.ts:48]: There are cy custom commands for handling this.
using a bunch of cy command like this is going to make upgrading to PF5 hard.

File: frontend/awx/resources/templates/WorkflowJobTemplateForm.tsx [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        job_tags: jobTags ?? '',
+        skip_tags: skipTags ?? '',
+        webhook_credential: values.summary_fields.webhook_credential?.id,
+      });
+      (cache as unknown as { clear: () => void }).clear?.();
+      const promises = [];
+
+      promises.push(submitLabels(workflowJobTemplate as WorkflowJobTemplate, labels?.results));
+      navigate(RouteObj.WorkflowJobTemplateDetails.replace(':id', `${id}`.toString()));
+    } catch (err) {
```
Comment (jamestalton) [inline frontend/awx/resources/templates/WorkflowJobTemplateForm.tsx:89]: should not need this catch. the form will catch it and display the error.

File: frontend/awx/resources/templates/WorkflowJobTemplateForm.tsx [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+      const promises = [];
+
+      if (labels?.results && labels?.results?.length > 0) {
+        promises.push(submitLabels(template, labels.results));
+      }
+
+      if (promises.length > 0) await Promise.all(promises);
+
+      navigate(RouteObj.WorkflowJobTemplateDetails.replace(':id', template.id.toString()));
+    } catch (err) {
```
Comment (jamestalton) [inline frontend/awx/resources/templates/WorkflowJobTemplateForm.tsx:167]: same. i do not think we should have to have this catch as the form handles it

**Review by jamestalton — CHANGES_REQUESTED** [review]

File: frontend/awx/resources/templates/WorkflowJobTemplateInputs.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        options={workflowJobTemplate?.arrayedSkipTags ?? [{ value: '', label: '' }]}
+      />
+      <PageFormCheckbox<WorkflowJobTemplateFormType>
+        label={t('Enable webhook')}
+        name="isWebhookEnabled"
+      />
+      <PageFormCheckbox<WorkflowJobTemplateFormType>
+        label={t('Enable Concurrent jobs')}
+        name="allow_simultaneous"
+      />
```
Comment (marshmalien) [inline frontend/awx/resources/templates/WorkflowJobTemplateInputs.tsx:120]: Could we move these two checkboxes into a separate form section? When the form is in 3 columns, the checkboxes get split up. 
Screenshot 2023-09-01 at 12 07 50 PM

File: frontend/awx/resources/templates/WorkflowJobTemplateForm.tsx [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+      skipTags = stringifyTags(arrayedSkipTags);
+    }
+
+    const template = await postRequest(`/api/v2/workflow_job_templates/`, {
+      ...values,
+      job_tags: jobTags,
+      skip_tags: skipTags,
+      summary_fields: values.summary_fields,
+      webhook_credential: webhook_credential?.id ? webhook_credential?.id : null,
+    });
```
Comment (marshmalien) [inline frontend/awx/resources/templates/WorkflowJobTemplateForm.tsx:148]: We don't need to send `summary_fields` on workflow template creation. I think the API is expecting a payload that looks something like this: 
```
interface WFCreation {
  allow_simultaneous: boolean;
  ask_inventory_on_launch: boolean;
  ask_labels_on_launch: boolean;
  ask_limit_on_launch: boolean;
  ask_scm_branch_on_launch: boolean;
  ask_skip_tags_on_launch: boolean;
  ask_tags_on_launch: boolean;
  ask_variables_on_launch: boolean;
  description: string;
  extra_vars: string;
  inventory: number;
  job_tags: string;
  limit: string;
  name: string;
  organization: number;
  scm_branch: string;
  skip_tags: string;
  survey_enabled: boolean;
  webhook_credential: number;
  webhook_service: 'github' | 'gitlab' | '';
}
```

**Review by marshmalien — CHANGES_REQUESTED** [review]

File: frontend/awx/resources/templates/WorkflowJobTemplateForm.tsx [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+    // eslint-disable-next-line no-useless-catch
+    try {
+      const data = await requestGet<AwxItemsResponse<Organization>>('/api/v2/organizations/');
+      orgId = data.results[0].id;
+    } catch (err) {
+      throw err;
+    }
+  }
+  const disassociationPromises = removed.map((label: { id: number }) =>
+    postRequest(`/api/v2/worflow_job_templates/${template.id.toString()}/labels/`, {
```
Comment (marshmalien) [inline frontend/awx/resources/templates/WorkflowJobTemplateForm.tsx:198]: Workflow labels URL spelling 
`/api/v2/workflow_job_templates/...`

**Review by marshmalien — APPROVED** [review]
Looks great! Nice work on the forms 🔥

**Review by vishsanghishetty — APPROVED** [review]

File: cypress/support/component.tsx [hunk 7]
```diff
@@ -63,8 +64,8 @@ declare global {
   }
 }
 
-Cypress.Commands.add('mount', (component, route) => {
-  cy.fixture('activeUser').then((activeUser: User) => {
+Cypress.Commands.add('mount', (component, route, fixture) => {
```
Comment (jamestalton) [inline cypress/support/component.tsx:67]: fixture is too generic here. it's the fixture for the active user correct?
maybe activeUserFixture?

**Review by jamestalton — APPROVED** [review]
lgtm

**Pull request merged**